eoscompanion/ekn_data.py

```diff
@@ -39,28 +55,30 @@ def load_record_blob_from_engine(engine, app_id, content_id, attr):
     Returns the a tuple of a (status code, blob) on success,
     (status code, None) otherwise.
     '''
-    if attr not in ('data', 'metadata'):
-        raise RuntimeError('attr must be one of "data" or "metadata"')
-
     try:
-        domain = engine.get_domain_for_app(app_id)
+        shards = engine.get_domain_for_app(app_id).get_shards()
     except GLib.Error as error:
         if (error.matches(Gio.io_error_quark(), Gio.IOErrorEnum.FAILED) or
                 error.matches(Gio.io_error_quark(), Gio.IOErrorEnum.NOT_FOUND)):
             return LOAD_FROM_ENGINE_NO_SUCH_APP, None
         raise error
 
-    shards = domain.get_shards()
+    return load_record_blob_from_shards(shards, content_id, attr)
 
-    for shard in shards:
-        record = shard.find_record_by_hex_name(content_id)
 
-        if not record:
-            continue
+def load_record_from_shards_async(shards, content_id, attr, callback):
```

Comment:
It's quite odd to have an async function here that only executes the callback in certain cases, and uses the return value for others. I would probably change it to always go through the callback.

Reply:
You have a good point here, this function is a little awkward. Only thing is that the callback is a GAsyncReady style callback so I'll need to convert it I suppose.